aquarius/events/events_monitor.py

```diff
@@ -265,17 +265,29 @@ def handle_token_uri_update(self, from_block, to_block):
                 )
 
     def get_last_processed_block(self):
-        block = 0
         try:
+            # Initialize the cached block with 0 if cache is empty.
+            if get_cached_block() is None:
```

Comment:
should the cached block be updated when the ES one is updated? i.e. when doing "store_last_processed_block"?
what happens if the REDIS connection is undefined? Does this code still work as expected (no caching, starts from 0)?

don't you think the existence of REDIS connection for block caching forces the nonce to use REDIS instead of sqlite? What if the user wants to define REDIS just for block caching, but wants to keep sqlite nonce generation? Is that a possibility?